### ISSUE TITLE
pep tries to find subservice even if no fiware-servicepath header was provided

### ISSUE THREAD

**Issue by AlvaroVega** [telefonicaid/fiware-orion-pep#165]
Test case: 
pep (listening in 7070 port) for keypass (listerning in 8080)

``` bash
SUB_SERVICE_ADMIN_ROLE_NAME=SubServiceAdmin
NEW_SERVICE_NAME=SmartValencia
NEW_SUBSERVICE_NAME=Electricidad

KEYSTONE_PROTOCOL=http
KEYPASS_PROTOCOL=http
KEYSTONE_ENDPOINT=localhost:5000
KEYPASS_ENDPOINT=localhost:7070


CLOUD_ADMIN_TOKEN=$(\
curl --insecure ${KEYSTONE_PROTOCOL}://${KEYSTONE_ENDPOINT}/v3/auth/tokens \
    -s \
    -i \
    -H "Content-Type: application/json" \
    -d '
{
  "auth": {
    "identity": {
      "methods": [
        "password"
      ],
      "password": {
        "user": {
          "domain": {
            "name": "admin_domain"
          },
          "name": "cloud_admin",
          "password": "password"
        }
      }
   },
   "scope": {
     "domain": {
       "name": "admin_domain"
     }
   }
  }
}' | grep ^X-Subject-Token: | awk '{print $2}' )
echo $CLOUD_ADMIN_TOKEN

SERVICE_ADMIN_TOKEN=$(\
curl --insecure ${KEYSTONE_PROTOCOL}://${KEYSTONE_ENDPOINT}/v3/auth/tokens \
    -i \
    -s \
    -H "Content-Type: application/json" \
    -d '
{
    "auth": {
        "identity": {
            "methods": [
               "password"
            ],
            "password": {
                "user": {
                    "domain": {
                        "name": "'$NEW_SERVICE_NAME'"
                    },
                    "name": "adm1",
                    "password": "password"
                }
            }
        },
        "scope": {
            "domain": {
                "name": "'$NEW_SERVICE_NAME'"
            }
        }
    }
}' | grep ^X-Subject-Token: | awk '{print $2}' )
echo $SERVICE_ADMIN_TOKEN


ID_DOM1=$(\
curl --insecure ${KEYSTONE_PROTOCOL}://${KEYSTONE_ENDPOINT}/v3/domains \
    -s \
    -H "X-Auth-Token: $CLOUD_ADMIN_TOKEN" \
    -H "Content-Type: application/json" \
| jq '.domains[] | select(.name == "'${NEW_SERVICE_NAME}'").id' | tr -d '"' )
echo "ID_DOM1=$ID_DOM1"

SUBSERVICE_ADMIN_ROLE_ID=$(\
curl --insecure ${KEYSTONE_PROTOCOL}://${KEYSTONE_ENDPOINT}/v3/OS-SCIM/Roles?domain_id=$ID_DOM1 \
    -s \
    -H "X-Auth-Token: $SERVICE_ADMIN_TOKEN" \
|  jq '.Resources[] | select(.name == "'${SUB_SERVICE_ADMIN_ROLE_NAME}'").id' | tr -d '"' )
echo "SUBSERVICE_ADMIN_ROLE_ID=$SUBSERVICE_ADMIN_ROLE_ID"

SERVICE_ADMIN_TOKEN=$(echo "$SERVICE_ADMIN_TOKEN" | tr -d '\r')
curl --insecure -i -H "Fiware-Service: $NEW_SERVICE_NAME" \
    -H "Content-type: application/xml" \
    -H "X-Auth-Token: $SERVICE_ADMIN_TOKEN" \
    ${KEYPASS_PROTOCOL}://$KEYPASS_ENDPOINT/pap/v1/subject/$SUBSERVICE_ADMIN_ROLE_ID

```

Response:
```
ID_DOM1=06b9a460f5c343b3bc8b43f8533f7ab4
SUBSERVICE_ADMIN_ROLE_ID=62c415f7e37147129d5951ee4da156f3
HTTP/1.1 403 Forbidden
X-Powered-By: Express
Unica-Correlator: 4d1ea904-69e0-471d-83bb-ff44a678806d
Content-Type: application/json; charset=utf-8
Content-Length: 79
ETag: "-392236945"
Date: Mon, 16 Feb 2015 08:53:17 GMT
Connection: keep-alive

{
  "message": "Could not find subservice with name [undefined] in Keystone."
}

```

Pep logs: 

```
time=2015-02-16T08:53:17.425Z | lvl=ERROR | corr=4d1ea904-69e0-471d-83bb-ff44a678806d | trans=4d1ea904-69e0-471d-83bb-ff44a678806d | op=/pap/v1/subject/62c415f7e37147129d5951ee4da156f3 | msg=Couldn't find subservice id in Keystone with name undefined
time=2015-02-16T08:53:17.426Z | lvl=ERROR | corr=4d1ea904-69e0-471d-83bb-ff44a678806d | trans=4d1ea904-69e0-471d-83bb-ff44a678806d | op=/pap/v1/subject/62c415f7e37147129d5951ee4da156f3 | msg=VALIDATION-GEN-003] Error connecting to Keystone authentication: KEYSTONE_SUBSERVICE_NOT_FOUND: Could not find subservice with name [undefined] in Keystone.
time=2015-02-16T08:53:17.427Z | lvl=DEBUG | corr=4d1ea904-69e0-471d-83bb-ff44a678806d | trans=4d1ea904-69e0-471d-83bb-ff44a678806d | op=/pap/v1/subject/62c415f7e37147129d5951ee4da156f3 | msg=response-time: 84 statusCode: 403

```

**Comment by fgalan** [comment]
Need to be clasified under the appropiated milestone depending on the importance of the bug. @jcalderin , could do that, please?

**Comment by fgalan** [comment]
Discussed by skype: it goes to 0.5.0.

**Comment by fgalan** [comment]
Fixed in PR #167 (for the records).
